lib/src/world.cairo

```diff
@@ -9,44 +10,52 @@ trait IProxy {
 
 #[abi]
 trait IWorld {
-    fn issue_entity(owner: starknet::ContractAddress) -> usize;
-    fn owner_of(entity_id: usize) -> starknet::ContractAddress;
-    fn entities(component: starknet::ContractAddress) -> Array<felt252>;
+    fn issue_entity(path: (felt252, felt252, felt252)) -> (felt252, felt252, felt252, felt252);
+    fn owner_of(entity_id: (felt252, felt252, felt252, felt252)) -> starknet::ContractAddress;
+    fn entities(component: starknet::ContractAddress) -> Array<(felt252, felt252, felt252, felt252)>;
 }
 
 trait ComponentTrait<T> {
     fn initialize();
-    fn set(entity_id: felt252, value: T);
-    fn get(entity_id: felt252) -> T;
+    fn set(entity_id: (felt252, felt252, felt252, felt252), value: T);
+    fn get(entity_id: (felt252, felt252, felt252, felt252)) -> T;
 }
 
-trait SystemTrait<T> {
-    fn execute(calldata: T);
+impl LegacyHashEntityRegsiteryTuple of LegacyHash::<(starknet::ContractAddress, felt252, felt252, felt252, felt252)> {
+    fn hash(state: felt252, tuple: (starknet::ContractAddress, felt252, felt252, felt252, felt252)) -> felt252 {
+        let (first, second, third, fourth, fifth) = tuple;
+        let mut state = LegacyHash::hash(state, first);
```

Comment:
I think it's possible to call `hash` on a 4-tuple. Not sure if it'll work with `first` being `ContractAddress`.

Reply:
Ah good point thanks!